packages/x-date-pickers-pro/src/DateRangePicker/date-range-manager.test.ts

```diff
@@ -3,9 +3,9 @@ import { calculateRangeChange, calculateRangePreview } from './date-range-manage
 import { adapterToUse } from '../../../../test/utils/pickers-utils';
```

Comment:
For some reason `yarn test:unit --date-adapter moment` doesn't work anymore:

```
$ yarn test:unit --date-adapter moment
yarn run v1.22.17
$ cross-env NODE_ENV=test node --expose_gc ./node_modules/.bin/mocha 'packages/**/*.test.{js,ts,tsx}' --exclude '**/node_modules/**' --timeout 3000 --date-adapter moment
including inaccessible elements by default
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Same issue with `luxon` adapter.

`date-fns` works fine, `day-js` also runs but some tests fail because `Date` is incompatible with the lib.

Reply:
> If it does not woks when passing the date as a props, it means we should add a parser somewhere in the codebase

Interesting, for some reason I thought that you should use date objects compatible with the adapter.
I've used this approach in https://github.com/mui/mui-x/issues/5000#issuecomment-1136316874 and it seemed to solve the issue

Reply:
Supporting other formats than the library of the adapter is a debated issue:

- **pro** : it's more flexible for developers (many issues are due to passing `Date` or `string` in props `value`, `minDate`, ...)
- **cons**: libraries are here to unify behavior across browsers, and the `onChange` return an adapter date object. So we encourage bad practice (dev should choose a library and only use it, not switching between the library and `Date`)

Reply:
But currently we only support library supported date, right?

Reply:
After doing the change from https://github.com/mui/mui-x/pull/5221#discussion_r903140891, the tests are running back again with different adapters. Not all tests are passing, there're two still failing, but they're also failing on `master`. I didn't investigate the cause but the error is "Mask does not support numbers with variable length such as 'M'.".

Reply:
Don't care, the reason is that the default format is not the same for all adapters, and some of them (thinking about luxon) use names instead of digits for months. Which is not supported by the `useMaskedInput`

Reply:
I've opened https://github.com/mui/mui-x/pull/5291 addressing this

Reply:
Ok, I'll not add the suggestion. The purpose here was just to fix the CI with the default adapter. Let #5291 to take care of this.